servers/src/main/java/tachyon/master/permission/FileSystemPermissionChecker.java

```diff
@@ -49,7 +50,63 @@ public static synchronized void initializeFileSystem(boolean permissionCheckEnab
   }
 
   /**
-   * This method checks access permission.
+   * Checks requested permission on the last node and basic permission on the whole path.
```

Comment:
update the argument `inodes` -> `path`.

If expose only one method, caller will have to cut the path if it want to check the parent dir. So wrap a method `checkParentOrAncestorPermission` to handle the cutting path work. Other logic is the same as `checkSelfPermission`.

Anyway, I could just keep one and let caller do that in next updates, if it is wanted.

checkOwner is actually not related currently. Remove it from patch to avoid confusion.